test/unit/specs/components/checkbox.spec.js

```diff
@@ -0,0 +1,31 @@
+import { render, screen } from '@testing-library/vue'
+import Checkbox from '~/components/Checkbox'
+import ByIcon from '~/assets/licenses/by.svg?inline'
+
+describe('Checkbox', () => {
+  it('should render a checkbox with a string label', () => {
+    render(Checkbox, { propsData: { id: 'simple', label: 'Is Code Poetry?' } })
+    // Get throws an error if element is not found, so this serves as an assert
+    screen.getByLabelText(/code/i, { checked: false })
+  })
+
+  it('should become checked when clicked', async () => {
+    render(Checkbox, { propsData: { id: 'simple', label: 'Is Code Poetry?' } })
+    const checkbox = screen.getByText(/code/i)
+    await checkbox.click()
+    screen.getByText(/code/i, { checked: true })
+  })
+
+  it('should render icons if provided in the default slot', async () => {
+    // @obulat: I cannot pass a required prop to `License` object to use it in the slot,
+    // nor can I check that svg is rendered using its accessible role. But if you use
+    // `screen.debug()`, you can see it is rendered :)
```

Comment:
Can we just use `querySelector` for `svg` on the container that `render` returns?

```js
const { container } = render(Checkbox, ...)

const icon = container.querySelector('svg')
```

Also for being able to render a full output with slots with props, I think it's appropriate to create a TestWrapper style component:

```js
const TestWrapper = Vue.component('TestWrapper', {
  template: `<Checkbox id="simple" label="Is Code Poetry?"><Licence :whatever="'foo'" /></Checkbox>`,
  components: { Checkbox, License },
})
```

That way you can render anything and everything that a regular Vue application would allow you to without having to juggle around the slots options that the `render` function gives you and its limitations.

Reply:
Thank you so much for this, @sarayourfriend ! I haven't seen `container` yet, and the `TestWrapper` seems just the solution I was looking for!